addons/mail/static/src/js/utils.js

```diff
@@ -53,10 +53,17 @@ function linkify(text, attrs) {
     attrs = _.map(attrs, function (value, key) {
         return key + '="' + _.escape(value) + '"';
     }).join(' ');
-    return text.replace(urlRegexp, function (url) {
+    var curIndex = 0;
+    var result = "";
+    var match;
+    while ((match = urlRegexp.exec(text)) !== null) {
+        result += _.escape(text.slice(curIndex, match.index));
+        var url = _.escape(match[0]);
         var href = (!/^https?:\/\//i.test(url)) ? "http://" + url : url;
-        return '<a ' + attrs + ' href="' + href + '">' + url + '</a>';
-    });
+        result += '<a ' + attrs + ' href="' + href + '">' + url + '</a>';
```

Comment:
In the same regard, `attrs` and `href` should be escaped differently than `url` here.

Reply:
I don't think we should rely on `"` not being part of the regex for security purposes. The escape functions are better being generic: inserting in an attribute defined with `"` should always escape that char, and inserting into node should always escape the tag brackets.

But according to my other comment, I'm actually wondering if using the `_` escape like before wouldn't be ok, if the problem is elsewhere anyway?